### PR TITLE
chore: upgrade go-libp2p

### PR DESCRIPTION
**Pull request by sukunrt**

### CODE REVIEW

**Review by vyzo — COMMENTED** [review]
this will have to wait until #553 is merged.

**Comment by vyzo** [comment]
unblock; #553 has merged.

**Review by Stebalien — APPROVED** [review]

**Review by vyzo — APPROVED** [review]

**Pull request merged**
